tools/codegen/api/cpp.py

```diff
@@ -273,10 +273,11 @@ def argument_faithful(
         return argument(a)
 
 def group_arguments(
```

Comment:
tbh this isn't really grouping arguments anymore, it's just flattening arguments into the appropriate order for cpp apis.

Reply:
true, that could be renamed. It't independent from this PR though. I'll delay this as part of the follow-up clean-up you were planning to get rid of the grouping classes as well.